### DISCUSSION TITLE
Very basic support of Dexcom (G6) with BlueZ

### DISCUSSION THREAD

**Discussion by belohoub** [NightscoutFoundation/xDrip#1946] [General]
Hello all,
my current hobby project is about creating a very basic native support for Dexcom G6 sensor in Linux (the sensor I currently use), actually to bring a native but very basic support (current bat/glycemia level values reading) to the desktop Linux, and also to the mobile OS I use ([ubports](https://ubports.com/)) - if successful I'll naturally release it.

I'm sorry, that my question is unrelated to your project, I'm only trying to create a minimal support for Dexcom G6 in parallel for a different platform and I would need some little help in understanding to the protocol and the setup and this is the place with experienced people :-)

I tried to go through your Dexcom G5/6 Android code, but I was unable to replicate the AUTH process for the Dexcom sensor with BlueZ. For the initial tries, I currently use bluetoothctl and gattlib. (I'm also considering QT as a vital and multi-platform alternative).

What I need is just to clarify the AUTH process, as my ability to understand your code is very limited and probably source of misinterpretation in my case. Could anyone provide a step-by-step description of the pairing and AUTH process for Dexcom G6 as a reference (scan-pairing-auth-rxData)?

Thank you!

**Comment by tzachi-dar** [comment]
I'm afraid that I don't know anything about g6 auth, but wanted to make sure you are aware of https://github.com/tzachi-dar/LibreAllHouse

This is a project that connects with miaomiao and libre2 and allow them to connect to xDrip.
Might help you if you decide that you want to connect to xDrip at some time.

**Reply by belohoub** [reply]
Thank you for the reply! However, I think, that this is something out of my current scope.

Simply said, I just want a "second display" for my current values, while practising BLE at the same time.

**Comment by tolot27** [comment]
You may find the [Dexcom-ESP32-Reader](https://github.com/TheEpicBigBoss/Dexcom-ESP32-Reader) repository helpful.

Alternatively, you can download the values from Nightscout, if your phone uploads it.

**Reply by belohoub** [reply]
This looks great for my purposes! I missed it, thank you!

**Comment by Navid200** [comment]
Let's say you succeed.
Are you going to use your Linux app only?
Or, do you intend to use it while using what you are using now at the same time?
You haven't mentioned what you are using now.  There may be limitations.

I understand that this is just a hobby.  But, it may help you if you consider what you will be (or won't be) able to do with it now before spending the time.

**Reply by belohoub** [reply]
Thank you for the reply!

To be more specific: I personally use a closed loop of Dexcom sensor with the Tandem pump, where Tandem pump is currently my only online display - and I plan to stay. I also use a recommended solution for offline data inspection and patient-doctor data sharing which is fine for me.

The purpose of the "application" should be (currently) just to provide the second display while staying "in" the solution provided by Dexcom/Tandem. The other reason to go through the implementation pain is to be motivated to go through BLE/BlueZ :-)

**Reply by Navid200** [reply]
Just in case you are unaware, xDrip can be used while the t:slim pump is also being used at the same time.
There are people who do that already.

**Reply by belohoub** [reply]
Thank you! I'm aware of it. The problem with xDrip is that it is not supported on my favourite platforms (while Android emulator is currently a great pain). 

I hope, that a compact C or QT console app supporting this particular sensor could be a benefit - not only for me, but also for few others :-)

**Comment by belohoub** [comment]
Just to archive things in this discussion. This thread is somehow related to #1220.

By @andremauricio4 and @davystrong
